### PR TITLE
Fix sparse_softmax_cross_entropy_ms subgraph

### PR DESCRIPTION
**Pull request by leaves-zwx**
修改 sparse_softmax_cross_entropy_ms 的前向子图构图。

原来的计算公式是 softmax + sparse_cross_entropy

现改为 log_softmax + nll

这么改的目的是为了增强数值的稳定性。（见 https://github.com/Oneflow-Inc/oneflow/issues/3137）

%20%5Cright%20)%7D%20%5C%5C&%20=%20-%20%5Clog_%7B%7D%7B%5Cfrac%7Be%5E%7Bx_%7Bi%7D%7D%7D%7B%20%7B%5Ctextstyle%20%5Csum_%7Bj%7D%5E%7B%7D%20%7D%20e%5E%7Bx_%7Bj%7D%7D%7D%7D%20%5C%5C%20&%20=%20-%20%5Clog_%7B%7D%7B%5Cfrac%7Be%5E%7Bx_%7Bi%7D%20-%20x_%7Bmax%7D%7D%7D%7B%7B%5Ctextstyle%20%5Csum_%7Bj%7D%5E%7B%7D%20%7D%20e%5E%7Bx_%7Bj%7D%20-%20x_%7Bmax%7D%7D%7D%7D%20%5C%5C%20&%20=%20-%20%5Clog_%7B%7D%7Be%5E%7Bx_%7Bi%7D%20-%20x_%7Bmax%7D%7D%7D%20&plus;%20%5Clog_%7B%7D%7B%7B%5Ctextstyle%20%5Csum_%7Bj%7D%5E%7B%7D%20%7D%20e%5E%7Bx_%7Bj%7D%20-%20x_%7Bmax%7D%7D%7D%20%5C%5C&%20=%20-%20%5Cleft%20(%20x_%7Bi%7D%20-%20x_%7Bmax%7D%20%5Cright%20)%20&plus;%20%5Clog_%7B%7D%7B%7B%5Ctextstyle%20%5Csum_%7Bj%7D%5E%7B%7D%20%7D%20e%5E%7Bx_%7Bj%7D%20-%20x_%7Bmax%7D%7D%7D%20%5C%5C%5Cend%7Balign*%7D)

### TODOs

- [x] 添加1个 sparse_softmax_cross_entropy_ms 的单测（以前 single-client 下存在但被删除）
- [ ]  (后续 pr) 修改 eager 下的 sparse_softmax_cross_entropy_ms 实现
- [ ]  (后续 pr) nn.CrossEntropyLoss 兼容 sparse_softmax_cross_entropy_ms
- [ ]  (后续 pr) nn.CrossEntropyLoss 兼容 sparse 和 probabilities 两种模式

### CODE REVIEW

**Review by liujuncheng — APPROVED** [review]

File: oneflow/core/job_rewriter/split_sparse_softmax_cross_entropy_op_pass.cpp [hunk 1]
```diff
@@ -188,19 +191,65 @@ Maybe<void> SplitSparseSoftmaxCrossEntropyOpPass::Apply(const OpGraph& op_graph,
...
+                      .Output("out")
+                      .Output("total_weight")
+                      .Attr<int64_t>("ignore_index", -100)
+                      .ScopeSymbolId(scope_symbol_id)
+                      .Build();
+    job_builder->AddOps(node->parallel_desc().parallel_conf(), {nll_op.op_conf()});
+
+    const std::string& prob_lbn = cur_op.output("prob", 0);
+    const std::string& out_lbn = cur_op.output("out", 0);
+    const std::string& new_prob_lbn = broadcast_div_op.output("z", 0);
```
Comment (guo-ran) [inline oneflow/core/job_rewriter/split_sparse_softmax_cross_entropy_op_pass.cpp:225]: 这里传给后向用的prob仍然是原来的softmax公式算出的结果。所以只是loss不正常，后向没有问题？

Comment (leaves-zwx) [reply oneflow/core/job_rewriter/split_sparse_softmax_cross_entropy_op_pass.cpp:225]: xi - xmax 负的特别大的时候 softmax 可能得到一个非常接近 0 的浮点数，然后 log(x) 得到 -inf，通过 safe_log clamp 到 1e-20，但仍然损失了精度。

但这里 prob 并没经过 log(x) 计算，所以后向不存在这个问题吧。

**Review by guo-ran — APPROVED** [review]
